data/src/main/java/com/microsoft/azure/kusto/data/BaseClient.java

```diff
@@ -37,79 +50,84 @@ public BaseClient(HttpClient httpClient) {
         this.httpClient = httpClient;
     }
 
-    protected String post(HttpRequest request, long timeoutMs) throws DataServiceException {
-        // Execute and get the response
-        try (HttpResponse response = httpClient.sendSync(request, getContextTimeout(timeoutMs))) {
-            return processResponseBody(response);
-        } catch (DataServiceException e) {
-            throw e;
-        } catch (Exception e) {
-            throw ExceptionUtils.createExceptionOnPost(e, request.getUrl(), "sync");
-        }
+    protected Mono<String> postAsync(HttpRequest request, long timeoutMs) {
+        return httpClient.send(request, getContextTimeout(timeoutMs))
+                .flatMap(response -> Utils.getResponseBody(response)
+                        .flatMap(responseBody -> {
```

Comment:
So my reasoning for moving to unchecked exceptions was that now we can use the light weight map instead of flat map for non-async calls - here is a good place for that - instead of using monos we can just throw exception

Reply:
Totally missed it. Fixed!